### DISCUSSION TITLE
Addtional registry urls with authentication token in URL

### DISCUSSION THREAD

**Discussion by y0x** [renovatebot/renovate#21814] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate should look up dependencies in two registries, where the access token is evaluted during pipeline execution and must be passed in the registry URL, like `https://aws:$CODEARTIFACT_TOKEN@company-*.d.codeartifact.eu-central-1.amazonaws.com/pypi/my-registry/simple/`

### What you tried so far.

1. Evaluate and set registry URL with correct access token during pipeline execution

```
.get_codeartifact_token:
  script:
    - echo Retrieve CodeArtifact Token
    - export CODEARTIFACT_TOKEN=`aws codeartifact get-authorization-token * * * --query authorizationToken --output text`
    - echo "Codeartifact-Token retrieved $CODEARTIFACT_TOKEN"
    - export PIP_INDEX_URL=https://aws:$CODEARTIFACT_TOKEN@*-*.d.codeartifact.eu-central-1.amazonaws.com/pypi/my-registry-1/simple/
    - export PIP_EXTRA_INDEX_URL=https://aws:$CODEARTIFACT_TOKEN@*-*.d.codeartifact.eu-central-1.amazonaws.com/pypi/my-registry-1/simple/

```
* = Privacy replacments

Getting a token works and also looking up dependencies from the PIP_INDEX_URL are working, but the PIP_EXTRA_INDEX_URL is ignored when ONLY setting them here.

From my understanding, I cannot define the URL's in renovate.json as the access token must be evaluated first (https://docs.renovatebot.com/python/#alternate-registries). I´ve also tested this, but no success. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by augi** [comment]
I can observe the same behavior - the `PIP_INDEX_URL` is used, but `PIP_EXTRA_INDEX_URL` is not checked.
